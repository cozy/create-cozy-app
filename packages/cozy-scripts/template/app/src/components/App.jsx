import React from 'react'
import { Route, Switch, Redirect, HashRouter } from 'react-router-dom'
import { hot } from 'react-hot-loader'

import { Layout, Main, Content } from 'cozy-ui/transpiled/react/Layout'
import { Sprite as IconSprite } from 'cozy-ui/transpiled/react/Icon'

import Sidebar from './Sidebar'
import Todos from './Todos'
import Hello1 from './HelloViews/Hello1'
import Hello2 from './HelloViews/Hello2'

const App = () => (
  <HashRouter>
    <Layout>
      <Sidebar />
      <Main>
        <Content className="app-content">
          <Switch>
            <Route path="/todos" component={Todos} />
            <Route path="/viewhello1" component={Hello1} />
            <Route path="/viewhello2" component={Hello2} />
            <Redirect from="/" to="/todos" />
            <Redirect from="*" to="/todos" />
          </Switch>
        </Content>
      </Main>
      <IconSprite />
    </Layout>
  </HashRouter>
)

/*
  Enable Hot Module Reload using `react-hot-loader` here
  We enable it here since App is the main root component
  No need to use it anywhere else, it sould work for all
  child components
*/
export default hot(module)(App)
