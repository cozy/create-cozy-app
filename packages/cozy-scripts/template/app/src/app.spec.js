'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import App from '../src/components/App'

describe('App component only', () => {
  it('should be mounted correctly', () => {
    const component = shallow(<App />).getElement()
    expect(component).toMatchSnapshot()
  })
})
