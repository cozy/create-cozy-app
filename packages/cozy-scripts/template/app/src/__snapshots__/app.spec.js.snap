// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component only should be mounted correctly 1`] = `
<HashRouter>
  <Layout
    monoColumn={false}
  >
    <withI18n(Sidebar) />
    <Main>
      <Content
        className="app-content"
      >
        <Switch>
          <Route
            component={[Function]}
            path="/todos"
          />
          <Route
            component={[Function]}
            path="/viewhello1"
          />
          <Route
            component={[Function]}
            path="/viewhello2"
          />
          <Redirect
            from="/"
            push={false}
            to="/todos"
          />
          <Redirect
            from="*"
            push={false}
            to="/todos"
          />
        </Switch>
      </Content>
    </Main>
    <IconSprite />
  </Layout>
</HashRouter>
`;
